PayloadPkg/OsLoader/OsLoader.c

```diff
@@ -304,16 +319,21 @@ SetupBootImage (
     return EFI_OUT_OF_RESOURCES;
   }
 
-  Status = EFI_SUCCESS;
   CmdFile = &LoadedImage->Image.Common.CmdFile;
-  if (CmdFile->Size > 0) {
-    CmdFile->Size = GetFromConfigFile (NewCmdBuffer, CMDLINE_LENGTH_MAX, (UINT8 *)CmdFile->Addr, CmdFile->Size);
+  Size = 0;
+  if (CmdFile->Addr != NULL && CmdFile->Size != 0) {
+    Size = GetFromConfigFile (NewCmdBuffer, CMDLINE_LENGTH_MAX, (UINT8 *)CmdFile->Addr, CmdFile->Size);
   }
-  if (CmdFile->Size == 0) {
+  if (Size == 0) {
     CopyMem (NewCmdBuffer, DEFAULT_COMMAND_LINE, AsciiStrSize (DEFAULT_COMMAND_LINE));
-    CmdFile->Size = AsciiStrSize (DEFAULT_COMMAND_LINE);
+    Size = AsciiStrSize (DEFAULT_COMMAND_LINE);
```

Comment:
Move #329 to before #328.  Then   CopyMem (NewCmdBuffer, DEFAULT_COMMAND_LINE, Size)

Reply:
Updated according to review comment.